### ISSUE TITLE
Cannot get it working

### ISSUE THREAD

**Issue by stefan-wiesner** [bmatcuk/html-brunch-static#9]
Hi.

I am stuck for hours trying to bring this baby up and running. No chance. 

I need your help to get what I am doing wrong.

[https://gist.github.com/stefan-wiesner/4e14f3ec5531360006c7bec034bfa39f](url)

Please check this gist where you can see what files I have.

They are set up according to your gist.

/app/index.static.hbs
/app/layouts/main.static.hbs
/app/partials/teaser.static.hbs

First of all I always get the error that the partial teaser cannot be found. But its defined in the index.static.hbs in front matter. Don't see why its not working.

And also the layout is at no point of time taken into the build process. For me it looks more like the layouts and the partials folder are completely ignoren than defined in the config file...

Then when I do a build the files are not created in the public folder that I am ending up with /myapp/layouts/main.static and an /myapp/index.static file... So path is wrong and the fileTransform is not fired. 

I dont have any clue what I am doing wrong. Could you please have a look and point me into the right direction?

Thanks

Stefan

**Comment by bmatcuk** [comment]
Hi Stefan,

I've found a couple issues with your `brunch-config.coffee`: there were three places where you put regex's inside single quotes so they're treated as strings and won't match anything. Also, your "does not fire" comment was syntactically incorrect (`//` is integer division in coffeescript so `// does not fire` was giving me the error: `does is not defined`). Changing `//` to `#` worked. Changes highlighted below:

```coffeescript
module.exports =
  paths:
    public: 'public'
    watched: [ 'app' ]
  server:
    hostname: '0.0.0.0'
    port: 3333
  files:
    javascripts: joinTo:
      'vendor.js': /^(?!app)/
      'app.js': /^app/
    stylesheets: joinTo: 'app.css'
    templates: joinTo: 'app.js'
  plugins:
    autoReload: enabled: true
    static: processors: [ require('html-brunch-static')(
      partials: /partials?/   # remove single quotes
      layouts: /layouts?/   # remove single quotes
      defaultContext: {}
      handlebars: enableProcessor:
        fileMatch: /.static.(hbs|handlebars)$/   # remove single quotes
        fileTransform: (f) ->
          console.log 'ooooooooooooooooooooooooooooooooooooooooo: ' + f   # does fire now, hopefully
          f.replace /\.static\.\w+$/, '.html'
    ) ]
```

With these changes, it's working for me with one exception: I intended partials to be sort of like "functions" in that they operate on the context already set. This means that they ignore any variables in the front matter inside the partial itself (so the `people` variable you're setting inside your partial is ignored and the `{{#each people}}` block outputs nothing). So you could get it working with something like:

```handlebars
---
_options:
  partials:
    - app/partials/teaser.static.hbs
people:
  - name: Devin
  - name: Kevin
---
{{#each people}}{{>teaser}}{{/each}}
```

where `teaser.static.hbs` is just:

```handlebars
Hello, {{name}}!
```

If you'd like to be able to set variables inside the partial itself, let me know and I can look into adding that as a feature... but it's a little bit complicated...

Anyway, if it still isn't working for you, what version of brunch are you using? Could you include your package.json?

-Bob

**Comment by stefan-wiesner** [comment]
Hi Bob.

Thanks a lot for the quick response. Yeah I messed it up with the quotes whiles changing from regex to path and back. Haven't seen it at all. Thanks a lot.

But is still wasn't working. And thanks to your request of the package.json I did a fresh startover from your gist [https://gist.github.com/bmatcuk/5fddb13bcf47d5616a5ba2b60caaf53b](url) and thats working perfectly fine now. 

Thank you so much. You saved my day!
